src/vehicle_use.cpp

```diff
@@ -1970,6 +1974,31 @@ void vehicle::use_bike_rack( int part )
     }
 }
 
+/*
+* Todo: find a way to split and rewrite use_bikerack so that this check is no longer necessary
+*/
+std::vector<std::vector<int>> vehicle::validate_carried_vehicles( std::vector<std::vector<int>>
+                           &carried_vehicles )
+{
+    std::sort( carried_vehicles.begin(), carried_vehicles.end(), []( const std::vector<int> &a,
+    const std::vector<int> &b ) {
+        return a.size() < b.size();
+    } );
+
+    std::vector<std::vector<int>>::iterator it = carried_vehicles.begin();
+    while( it != carried_vehicles.end() ) {
+        for( std::vector<std::vector<int>>::iterator it2 = it + 1; it2 < carried_vehicles.end(); it2++ ) {
+            if( std::search( ( *it2 ).begin(), ( *it2 ).end(), ( *it ).begin(),
+                             ( *it ).end() ) != ( *it2 ).end() ) {
+                it = carried_vehicles.erase( it );
+            }
+        }
+        it++;
+    }
+    return carried_vehicles;
```

Comment:
Now you are modifying the parameter reference in-place so you don't need return value anymore.

Reply:
updated